pkg/recipes/terraform/install.go

```diff
@@ -78,5 +81,40 @@ func Install(ctx context.Context, installer *install.Installer, tfDir string) (s
 
 	logger.Info(fmt.Sprintf("Terraform latest version installed to: %q", execPath))
 
-	return execPath, nil
+	// Create a new instance of tfexec.Terraform with current Terraform installation path
+	tf, err := NewTerraform(ctx, tfDir, execPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify Terraform installation is complete before proceeding
+	for attempt := 1; attempt <= installVerificationRetryCount; attempt++ {
+		_, _, err = tf.Version(ctx, false)
+		if err == nil {
+			metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
+				[]attribute.KeyValue{
+					metrics.TerraformVersionAttrKey.String("latest"),
+					metrics.OperationStateAttrKey.String(metrics.SuccessfulOperationState),
+				},
+			)
+			break
+		}
+		if attempt < installVerificationRetryCount {
+			logger.Info(fmt.Sprintf("Failed to verify Terraform installation completion: %s. Retrying after %d seconds", err.Error(), installVerificationRetryDelaySecs))
+			metrics.DefaultRecipeEngineMetrics.RecordTerraformInstallVerificationDuration(ctx, installStartTime,
```

Comment:
The last try will not be logged in this case.

Reply:
Case: Let's imagine that the **attempt** starts from 1 and the **installVerificationRetryCount** is 5. The fifth run will not go into this `if case` and therefore not be logged.

Reply:
Going to do it from 0 to retry-1 as suggested so there's `installVerificationRetryCount` logs

Reply:
Actually, there needs to be an attempt 5 for the if case to trigger in the case of an error. So I think the best way to do it is to stick to the original 0 to `installVerificationRetryCount`